packages/ui/src/components/DeleteConfirmationDialog/index.tsx

```diff
@@ -0,0 +1,174 @@
+/** *******************************************************************************************************************
+  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+  
+  Licensed under the Apache License, Version 2.0 (the "License").
+  You may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+  
+      http://www.apache.org/licenses/LICENSE-2.0
+  
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.                                                                              *
+ ******************************************************************************************************************** */
+
+import { ReactNode, FC, useCallback, useEffect, useMemo, useState } from 'react';
+import SpaceBetween from '@cloudscape-design/components/space-between';
+import Modal from '@cloudscape-design/components/modal';
+import FormField from '@cloudscape-design/components/form-field';
+import Input from '@cloudscape-design/components/input';
+import Button from '@cloudscape-design/components/button';
+import Box from '@cloudscape-design/components/box';
+
+export interface DeleteConfirmationWithFrictionProps {
+    variant?: 'friction';
+    /**
+     * The label of the input field.
+     */
+    label?: ReactNode;
+    /**
+     * The placeholder text of the input field.
+     */
+    placeholderText?: string;
+    /**
+     * The hint text of the input field.
+     */
+    hintText?: string;
+    /**
+     * The confirmation text that is expected from users to type in the input field.
+     */
+    confirmationText?: string;
+}
+
+export interface DeleteConfirmationWithConfirmationProps {
+    variant: 'confirmation';
+}
+
+export interface DeleteConfirmationBaseProps {
+    /**The variant of the Delete Confirmation Dialog*/
+    variant?: 'confirmation' | 'friction';
```

Comment:
maybe it will be better to provide enum for this constants?
```js
enum DeleteConfirmationVariant {
  CONFIRMATION = 'confirmation',
  FRICTION = 'friction',
}
```

Reply:
+1

Reply:
For React props, I prefer using union type than enum to avoid Import/Export. And this is also consistent with how Cloudscape is dealing with props like varaint.